core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -92,6 +96,19 @@ private void updateAdPlaybackState() {
         }
     }
 
+    public long advertDurationBy(int advertGroupIndex, int advertIndexInAdvertGroup) {
+        if (advertBreaks == null || advertGroupIndex >= advertBreaks.size()) {
+            return 0;
+        }
+
+        AdvertBreak advertBreak = advertBreaks.get(advertGroupIndex);
+        if (advertIndexInAdvertGroup >= advertBreak.adverts().size()) {
+            return 0;
```

Comment:
Looks like these two cases is actually an error, shall we throw an exception then? or `Log.e(...)`?

Reply:
Yes we probably could, it would be a fundamental flaw in ExoPlayer or our code if this happened.

Reply:
Throwing our runtime exception would be exactly the same behavior as if we didn't do any checks at all and we would just get the items from the lists directly. So wouldn't it be better to remove the checks and make it simpler? (Since this case should never happen)